enhancements/security/tls-artifacts-registry.md

```diff
@@ -0,0 +1,225 @@
+    for the lifecycle of this artifact
+* Create a test which saves existing information about TLS artifacts 
+  in cluster.
+  * Some secrets may be versioned, contain hash or IP address / DNS 
+    address - i.e. 
+    `etcd-peer-ip-10-10-10-10.us-east-1.internal` becomes 
+    `etcd-peer-for-master-2` etc.
+    The test would not store versioned copies and replace IPs with "master-n"
+    to make them look uniform when saving raw TLS information
+* Create a new test, which ensures that all secrets are present in the 
+  registry
+* Create a test that verifies that TLS artifacts in a cluster have 
+  necessary annotations to provide required metadata.
+* Some TLS artifacts can be added to known violation list so that the test 
+  is not permafailing to give teams time to update metadata. Each metadata 
+  property has its own violation list.
+* Metadata is split into required and optional properties. The test would 
+  record missing optional metadata and fail if required metadata is missing.
+  This gives time to update certificate metadata when optional metadata 
+  requirements become required. Violation files are meant to be "remove-only" 
+  mode, meaning when metadata property becomes required users are not meant 
+  to be added to the respective violation file.
+* Initially required metadata properties would be "owning JIRA component" and 
+  "one line description of artifact purpose". Later on other properties would be 
+  added and eventually become required for new certificates.
+* TLS artifact registry is being composed as a JSON file from several 
+  parts (usually generated from the test job artifacts).
+* In the `origin` repo create `hack/update-tls-artifacts.json` and 
+  `hack/verify-tls-artifacts.json` scripts to compose TLS artifact 
+  registry JSON from several JSONs and verify that all necessary 
+  metadata is set respectively. `update` script also generate violation lists
+  in JSON and Markdown formats.
+* Update library-go functions to set metadata when the TLS artifact is 
+  being generated, so that most operators would use the recommended 
+  method.
+
+### Workflow Description
+
+The cluster administrator opens the TLS artifacts registry in the `origin` repo 
+and consults it to ensure that a particular certificate is platform-managed. 
+The administrator can use JIRA component metadata to file a 
+new issue related to this certificate.
+
+When a components requires a new certificate created in cluster, . In order to properly 
+register a new TLS certificate the developer should do the following:
+
+* the developers should start create a pull request adding this functionality.
+  "all tls certificates should be registered" test would fail. 
+  The test job generates `rawTLSInformation` file which 
+  contains collected information from the prow job, containing info about this new 
+  certificate.
+* the developer creates a PR to `origin` repo, adding this file to 
```

Comment:
Where will this process be formally documented? Is this enhancement the final doc, or is there a separate developer guide?

Reply:
This enhancement will have a high-level overview of the process, `origin` repo now has a [TLS artifacts README](https://github.com/openshift/origin/blob/master/tls/ownership/README.md)  about specifics. Failed test also links to this README to streamline the process

Reply:
Sounds good, thanks